### PR TITLE
AISW-24708 Update bike sharing example

### PR DESCRIPTION
**Pull request by Nishant-HPE**
- Example needed code refactoring to be compatible with mlfow 3.1.4
- Updated model logging logic
- removed charts logging logic

output screenshots:

image

image

### CODE REVIEW

**Review by akanshasajimon — APPROVED** [review]
lgtm

File: Data-Science/MLflow/bike-sharing-mlflow.ipynb [hunk 1]
```diff
@@ -546,8 +552,8 @@
     "    # Save model to artifacts\n",
     "    mlflow.sklearn.log_model(model, \"model\", signature=signature)\n",
     "\n",
-    "    # log charts\n",
-    "    mlflow.log_artifacts(\"model_artifacts\")\n",
+    "    # # log charts\n",
```
Comment (roopak-rastogi) [inline Data-Science/MLflow/bike-sharing-mlflow.ipynb]: nit: Line is already commented. Consider removing double commenting.

**Comment by roopak-rastogi** [comment]
There was some discussion about updating the kserve inference endpoint. Are those changes missing here?

File: Data-Science/MLflow/bike-sharing-mlflow.ipynb [hunk 2]
```diff
@@ -39,7 +39,13 @@
...
     "experiment_name = 'bike-sharing-exp'\n",
     "\n",
-    "mlflow.set_experiment(experiment_name)"
+    "mlflow.set_experiment(experiment_name)\n",
+    "experiment = mlflow.set_experiment(experiment_name)\n",
+    "artifact_location = experiment.artifact_location\n",
+    "exp_id = experiment.experiment_id\n",
+    "\n",
+    "print(artifact_location)\n",
+    "print(exp_id)"
```
Comment (roopak-rastogi) [inline Data-Science/MLflow/bike-sharing-mlflow.ipynb]: Consider adding a label to clarify the output.
As an example, `print(f"Artifact Location: {artifact_location}")`

**Comment by akanshasajimon** [comment]
> There was some discussion about updating the kserve inference endpoint. Are those changes missing here?

This line is the update :
`"best_model_uri = f\"{artifact_location}/models/{best_model.model_id}/artifacts\"\n",`

Just to clarify, the update was made to the S3 URI rather than the inference endpoint.

**Review by akumar-47 — APPROVED** [review]

**Comment by akumar-47** [comment]
@Nishant-HPE could you address the review comment made by @roopak-rastogi , we need this example to be in working condition in dev build.

**Comment by Nishant-HPE** [comment]
> @Nishant-HPE could you address the review comment made by @roopak-rastogi , we need this example to be in working condition in dev build.

Hi, resolved all of the above comments. Also updated storage URI as requested by @roopak-rastogi . Thanks.

**Review by akanshasajimon — APPROVED** [review]

**Review by roopak-rastogi — APPROVED** [review]

**Comment by roopak-rastogi** [comment]
If possible, please attach the screenshot of the prediction output and the kserve endpoint generated.

**Review by ask664 — APPROVED** [review]
